diff --git a/node_modules/app-builder-lib/out/appInfo.js b/node_modules/app-builder-lib/out/appInfo.js
index 25a159e..bfe0590 100644
--- a/node_modules/app-builder-lib/out/appInfo.js
+++ b/node_modules/app-builder-lib/out/appInfo.js
@@ -165,7 +165,7 @@ class AppInfo {
   get linuxPackageName() {
     const name = this.name; // https://github.com/electron-userland/electron-builder/issues/2963
 
-    return name.startsWith("@") ? this.sanitizedProductName : name;
+    return 'tabby-terminal';
   }
 
   get sanitizedName() {
