diff --git a/node_modules/app-builder-lib/out/appInfo.js b/node_modules/app-builder-lib/out/appInfo.js
index f241acc..2bddb7f 100644
--- a/node_modules/app-builder-lib/out/appInfo.js
+++ b/node_modules/app-builder-lib/out/appInfo.js
@@ -100,9 +100,7 @@ class AppInfo {
         return this.info.metadata.name;
     }
     get linuxPackageName() {
-        const name = this.name;
-        // https://github.com/electron-userland/electron-builder/issues/2963
-        return name.startsWith("@") ? this.sanitizedProductName : name;
+        return 'tabby-terminal'
     }
     get sanitizedName() {
         return sanitizeFileName_1.sanitizeFileName(this.name);
